### DISCUSSION TITLE
gyp error with better-sqlite3 starting from renovate version 37.160.0

### DISCUSSION THREAD

**Discussion by Apamobil** [renovatebot/renovate#27844] [Request Help]
### What would you like help with?

Resolving the gyp error with better-sqlite3

### How are you running Renovate?

None

### Please tell us more about your question or problem

I'm currently experiencing a problem with Renovatebot. In January they introduced a change in version 37.160.0 that broke the compatibility with our Jenkins setup and all the versions that are newer than that are also not working.

37.160.0 release note ( https://github.com/renovatebot/renovate/releases/tag/37.160.0 ) states as the only new feature the introduction of SQLite-based package cache.

In the Jenkins console logs I can see gyp errors connected with better-sqlite3 after running the npm install renovate@37.160.0 command (see logs).

There is no connectivity with internet from the Jenkins agents and thus, it is impossible to fetch https://nodejs.org/download/release/v20.11.1/node-v20.11.1-headers.tar.gz directly from the nodejs.org server. I have got a line for the npmrc registry in the config.js file but it doesn't seem to help with this:

// Language specific configurations
  // https://docs.renovatebot.com/getting-started/private-packages/#add-npmrc-string-to-renovate-config
  "npmrc": "registry=https://<myartifactoryinstance>.com/api/npm/npm-general",  
};

Can you suggest what could be tried out? Now I'm stuck with the version 37.159.1

### Logs (if relevant)

<details><summary>Logs</summary>

```


+ npm install renovate@37.160.0
npm ERR! code 1
npm ERR! path /home/jenkins/workspace/enovate_RENOTEST_RENOTEST_master/renovate/node_modules/better-sqlite3
npm ERR! command failed
npm ERR! command sh -c prebuild-install || node-gyp rebuild --release
npm ERR! prebuild-install warn install connect ECONNREFUSED 140.82.121.4:443
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@10.0.1
npm ERR! gyp info using node@20.11.1 | linux | x64
npm ERR! gyp info find Python using Python version 3.10.12 found at "/usr/bin/python3"
npm ERR! gyp http GET https://nodejs.org/download/release/v20.11.1/node-v20.11.1-headers.tar.gz
npm ERR! gyp WARN install got an error, rolling back install
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack FetchError: request to https://nodejs.org/download/release/v20.11.1/node-v20.11.1-headers.tar.gz failed, reason:
npm ERR! gyp ERR! stack at ClientRequest.<anonymous> (/home/jenkins/workspace/enovate_RENOTEST_RENOTEST_master/renovate/node_modules/minipass-fetch/lib/index.js:130:14)
npm ERR! gyp ERR! stack at ClientRequest.emit (node:events:518:28)
npm ERR! gyp ERR! stack at _destroy (node:_http_client:875:13)
npm ERR! gyp ERR! stack at onSocketNT (node:_http_client:895:5)
npm ERR! gyp ERR! stack at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
npm ERR! gyp ERR! System Linux 5.15.0-43-generic
npm ERR! gyp ERR! command "/ciop/tools/nodejs/node-v20.11.1-linux-x64/bin/node" "/home/jenkins/workspace/enovate_RENOTEST_RENOTEST_master/renovate/node_modules/.bin/node-gyp" "rebuild" "--release"
npm ERR! gyp ERR! cwd /home/jenkins/workspace/enovate_RENOTEST_RENOTEST_master/renovate/node_modules/better-sqlite3
npm ERR! gyp ERR! node -v v20.11.1
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /tmp/npm/cache/_logs/2024-03-11T08_21_05_584Z-debug-0.log


```

</details>

**Comment by rarkins** [comment]
Are you npm installing renovate? If so then try running a containerized version

**Reply by Apamobil** [reply]
Yes, I'm npm installing it. Is npm installation deprecated? Anyway, I will try the containerized version. Thank you!

**Reply by rarkins** [reply]
It's not deprecated

**Reply by viceice** [reply]
> It's not deprecated

but needs Internet access for various reasons.
